arch/riscv/core/isr.S

```diff
@@ -388,28 +409,38 @@ is_user_syscall:
 	lr a3, __z_arch_esf_t_a3_OFFSET(sp)
 	lr a4, __z_arch_esf_t_a4_OFFSET(sp)
 	lr a5, __z_arch_esf_t_a5_OFFSET(sp)
+	lr CALL_ID_REG, __CALL_ID_REG_OFFSET(sp)
+#if defined(CONFIG_RISCV_ISA_RV32E)
+	addi sp, sp, -4
```

Comment:
Shouldn't `sp` be 16-byte aligned?

Reply:
Alignment for RV32E is down to 4 bytes. 16 bites only for RV32I.

Reply:
Oh, cool. Maybe add a comment here, as there are other places with comments stating the 16-byte alignment?

Reply:
Added a comment a few lines above